project-base/storefront/components/PaymentsInOrderSelect/PaymentsInOrderSelect.tsx

```diff
@@ -0,0 +1,123 @@
+import { InfoIcon, SpinnerIcon } from 'components/Basic/Icon/IconsSvg';
+import { Button } from 'components/Forms/Button/Button';
+import { Radiobutton } from 'components/Forms/Radiobutton/Radiobutton';
+import { GoPayGateway } from 'components/Pages/Order/PaymentConfirmation/Gateways/GoPayGateway';
+import { TransportAndPaymentListItem } from 'components/Pages/Order/TransportAndPayment/TransportAndPaymentSelect/TransportAndPaymentListItem';
+import { TransportAndPaymentSelectItemLabel } from 'components/Pages/Order/TransportAndPayment/TransportAndPaymentSelect/TransportAndPaymentSelectItemLabel';
+import { useOrderAvailablePaymentsQueryApi } from 'graphql/generated';
+import { useChangePaymentInOrder } from 'hooks/order/useChangePaymentInOrder';
+import useTranslation from 'next-translate/useTranslation';
+import { useEffect, useState } from 'react';
+import { PaymentTypeEnum } from 'types/payment';
+
+type PaymentsInOrderSelectProps = {
+    orderUuid: string;
+    orderPaymentType: string | undefined;
+    canPaymentBeRepeated: boolean;
+};
+
+export const PaymentsInOrderSelect: FC<PaymentsInOrderSelectProps> = ({
+    orderUuid,
+    canPaymentBeRepeated,
+    orderPaymentType,
+}) => {
+    const { t } = useTranslation();
+    const { changePaymentInOrderFetching, changePaymentInOrderHandler } = useChangePaymentInOrder();
+    const [selectedPaymentUuid, setSelectedPaymentUuid] = useState<string | undefined>();
+
+    const [{ data: orderAvailablePaymentsData, fetching: orderAvailablePaymentsFetching }] =
+        useOrderAvailablePaymentsQueryApi({
+            variables: { orderUuid },
+        });
+
+    const currentOrderPayment = orderAvailablePaymentsData?.orderPayments.currentPayment;
+
+    useEffect(() => {
+        setSelectedPaymentUuid(currentOrderPayment?.uuid);
+    }, [currentOrderPayment?.uuid]);
+
+    if (orderAvailablePaymentsFetching) {
+        return <SpinnerIcon className="mx-auto mt-4 block w-12" />;
+    }
+
+    if (!orderAvailablePaymentsData) {
+        return null;
+    }
+
+    return (
+        <div className="mt-6 flex flex-col items-center gap-6">
+            {orderPaymentType === PaymentTypeEnum.GoPay && canPaymentBeRepeated && currentOrderPayment && (
+                <div className="flex w-full flex-col items-center">
+                    <TransportAndPaymentListItem
+                        key={currentOrderPayment.uuid}
+                        className="order-none"
+                        isActive={selectedPaymentUuid === currentOrderPayment.uuid}
```

Comment:
The naming of "selected" and "current" makes it seems like it is the same thing and might be somewhat confusing. Could you consider different naming?